Makefile

```diff
@@ -30,35 +30,16 @@ DEVWORKSPACE_CTRL_SA=devworkspace-controller-serviceaccount
 INTERNAL_TMP_DIR=/tmp/devworkspace-controller
 BUMPED_KUBECONFIG=$(INTERNAL_TMP_DIR)/kubeconfig
 RELATED_IMAGES_FILE=$(INTERNAL_TMP_DIR)/environment
+GO_PACKAGE_PATH ?= $(shell head -n1 go.mod | cut -d " " -f 2)
+BUILD_TIME = $(shell date -u '+%Y-%m-%dT%H:%M:%SZ')
+REF = $(shell cat .git/HEAD | sed 's|ref: ||')
+GIT_COMMIT_ID := $(shell cat .git/${REF})
 
-ifeq (,$(shell which kubectl))
-ifeq (,$(shell which oc))
-$(error oc or kubectl is required to proceed)
-else
-K8S_CLI := oc
-endif
-else
-K8S_CLI := kubectl
-endif
-
-ifeq ($(shell $(K8S_CLI) api-resources --api-group='route.openshift.io'  2>&1 | grep -o routes),routes)
-PLATFORM := openshift
-else
-PLATFORM := kubernetes
-endif
-
-# minikube handling
-ifeq ($(shell $(K8S_CLI) config current-context 2>&1),minikube)
-# check ingress addon is enabled
-ifeq ($(shell minikube addons list -o json | jq -r .ingress.Status), disabled)
-$(error ingress addon should be enabled on top of minikube)
-endif
-export ROUTING_SUFFIX := $(shell minikube ip).nip.io
+ifneq (,$(shell which kubectl 2>/dev/null)$(shell which oc 2>/dev/null))
+include deploy.mk
 endif
```

Comment:
The check for `oc/kubectl` is necessary as most rules require it in some way or another. This should either always include `deploy.mk` and let it fail if kubectl/oc are not installed, or catch this case with an else:
```makefile
ifneq (,$(shell which kubectl 2>/dev/null)$(shell which oc 2>/dev/null))
include deploy.mk
else
$(error either oc or kubectl must be installed)
endif
```

Reply:
Ohh I see the problem, in the dockerfile we don't have `oc` and `kubectl` installed. Hmmm -- the only way I see around this that keeps the user-friendly(-ish) error message is something like a hidden target
```makefile
_check_oc_kubectl:
	if ! command -v kubectl &> /dev/null && ! command -v oc &> /dev/null; then \
		echo "Either kubectl or oc must be installed for this target" ;\
		exit 1 ;\
	fi
```
that all targets that use `oc`/`kubectl` depend on

Reply:
@amisevsk I've moved the other oc/kubectl targets to the deploy.mk - sorry for the confusion, should have been more clear on the split :woman_facepalming: